core/include/moveit/task_constructor/solvers/pipeline_planner.h

```diff
@@ -71,13 +71,16 @@ class PipelinePlanner : public PlannerInterface
 
 	/** \brief Constructor
 	 * \param [in] node ROS 2 node
-	 * \param [in] pipeline_id_planner_id_map map containing pairs of pipeline and plugin names to be used for planning
-	 * \param [in] stopping_criterion_callback callback to decide when to stop parallel planning
-	 * \param [in] solution_selection_function callback to choose the best solution from multiple ran pipelines
+	 * \param [in] pipeline_id_planner_id_map Map containing pairs of pipeline and plugin names to be used for planning
+	 * \param [in] planning_pipelines Optional: A map with the names and initialized corresponding planning pipelines
+	 * \param [in] stopping_criterion_callback Optional: Callback to decide when to stop parallel planning
+	 * \param [in] solution_selection_function Optional: Callback to choose the best solution from multiple ran pipelines
 	 */
 	PipelinePlanner(
 	    const rclcpp::Node::SharedPtr& node,
 	    const std::unordered_map<std::string, std::string>& pipeline_id_planner_id_map,
+	    const std::unordered_map<std::string, planning_pipeline::PlanningPipelinePtr>& planning_pipelines =
```

Comment:
What was you motivation to remove this argument?

Reply:
There were two/three conflicting approaches to instantiate the internal pipelines map:
1. By passing a map of instantiated pipelines via the "optional" `planning_pipelines` argument
2. By passing the `pipeline_id_planner_id_map`, which was used to create the former if not provided
3. A `pipeline_id_planner_id_map` property of the stage

In any case, the arguments `pipeline_id_planner_id_map` and `planning_pipelines` were designed to be _mutually exclusive_. This should have been implemented with two separate constructor methods to avoid inconsistencies.

Eventually, the property value was the driving factor for actual planning. Hence, I decided to just go for this property and - for convenience - initialize that property from the passed map argument.

I dropped the first option completely, because instantiating pipelines during stage construction is not trivial. It requires to pass the same `RobotModelPtr` that is used by the task. Having this option, puts you at risk to have a pipelines map that is inconsistent with the map property.

Reply:
Makes sense, thanks for that explanation!